vendors/espressif/boards/esp32/ports/ble/nimble/iot_ble_hal_common_gap.c

```diff
@@ -525,33 +525,53 @@ void ble_store_config_init( void );
 BTStatus_t prvBTManagerInit( const BTCallbacks_t * pxCallbacks )
 {
     BTStatus_t xStatus = eBTStatusSuccess;
+    esp_err_t xRet;
 
-    nimble_port_init();
 
-    ble_hs_cfg.reset_cb = bleprph_on_reset;
-    ble_hs_cfg.sync_cb = bleprph_on_sync;
-    ble_hs_cfg.store_status_cb = ble_store_util_status_rr;
-    ble_store_config_init();
+    xRet = esp_nimble_hci_and_controller_init();
 
-    if( pxCallbacks != NULL )
+    if( xRet == ESP_OK )
     {
-        xBTCallbacks = *pxCallbacks;
+        if( pxCallbacks != NULL )
+        {
+            xBTCallbacks = *pxCallbacks;
+        }
+        else
+        {
+            xStatus = eBTStatusFail;
+        }
     }
     else
     {
         xStatus = eBTStatusFail;
     }
 
+    if( xStatus == eBTStatusSuccess )
+    {
+        ble_hs_cfg.reset_cb = bleprph_on_reset;
+        ble_hs_cfg.sync_cb = bleprph_on_sync;
+        ble_hs_cfg.store_status_cb = ble_store_util_status_rr;
+
+        ble_store_config_init();
+    }
+
     return xStatus;
 }
 
 /*-----------------------------------------------------------*/
 
 BTStatus_t prvBtManagerCleanup()
 {
+    esp_err_t xRet;
     BTStatus_t xStatus = eBTStatusSuccess;
 
-    esp_bt_controller_mem_release( ESP_BT_MODE_BLE );
+    xRet = esp_nimble_hci_and_controller_deinit();
+
+    if( xRet != ESP_OK )
+    {
+        ESP_LOGE( TAG, "esp_nimble_hci_and_controller_deinit() failed with error %d", xRet );
```

Comment:
when to use ESP_LOGE and when to use configPrint?

Reply:
ESP_LOGE  is used since its the porting layer of ESP.